pkg/executor/types.go

```diff
@@ -16,30 +18,40 @@ type ExecutorProvider = model.Provider[model.Engine, Executor]
 type Executor interface {
 	model.Providable
 
-	// used to filter and select jobs
-	//    tells us if the storage resource is "close" i.e. cheap to access
-	HasStorageLocally(context.Context, model.StorageSpec) (bool, error)
-
 	// A BidStrategy that should return a positive response if the executor
 	// could run the job or a negative response otherwise.
 	GetSemanticBidStrategy(context.Context) (bidstrategy.SemanticBidStrategy, error)
 
 	GetResourceBidStrategy(ctx context.Context) (bidstrategy.ResourceBidStrategy, error)
 
-	//    tells us how much storage the given volume would consume
-	//    which we then use to calculate if there is capacity
-	//    alongside cpu & memory usage
-	GetVolumeSize(context.Context, model.StorageSpec) (uint64, error)
-
 	// GetOutputStream retrieves a muxed stream from the executor
 	GetOutputStream(ctx context.Context, executionID string, withHistory bool, follow bool) (io.ReadCloser, error)
 
 	// run the given job - it's expected that we have already prepared the job
 	// this will return a local filesystem path to the jobs results
-	Run(
-		ctx context.Context,
-		executionID string,
-		job model.Job,
-		resultsDir string,
-	) (*model.RunCommandResult, error)
+	Run(ctx context.Context, Args *RunCommandRequest) (*model.RunCommandResult, error)
+}
+
+type RunCommandRequest struct {
```

Comment:
Would that mean EnvironmentVariables becoming its own field on the JobSpec? Generally seems fine to me.

Reply:
I also like the idea of moving Envs outside of the engine spec to the `Job.Spec` level. My only concern would be if an executor and a publisher within the same spec want to use different values for the same env variable. We don't support Envs in publishers today, so it may not be an immediate concern